### ISSUE TITLE
Ignore windows types: dialog windows

### ISSUE THREAD

**Issue by alexjp** [esjeon/krohnkite#7]
Hi,

Could this script have an option to automatically ignore/set as floating by default windows of type: dialog ?

Swaywm for example, does this by default ( dialog windows don't get tiled ).

Thanks !

**Comment by esjeon** [comment]
My initial speculation is that filtering _NET_WM_STATE_MODAL (exposed as `modal` in [`KWin::Client`](https://techbase.kde.org/Development/Tutorials/KWin/Scripting/API_4.9#KWin::Client)), and adding a configuration option, enabled by default, should work.

**Comment by osleg** [comment]
Unfortunately dialog windows could be marked differently
Those could be `_NEW_WM_TRANSIET_FOR` clients and/or `_NET_WM_WINDOW_TYPE_DIALOG`.
Also `...._TYPE_UTILITY` usually should be marked as floated. Those could be a gimp brushes panel for example.

**Comment by osleg** [comment]
```
 _NET_WM_WINDOW_TYPE_UTILITY indicates a small persistent utility window, such as a palette or toolbox. It is distinct from type TOOLBAR because it does not correspond to a toolbar torn off from the main application. It's distinct from type DIALOG because it isn't a transient dialog, the user will probably keep it open while they're working. Windows of this type may set the WM_TRANSIENT_FOR hint indicating the main application window.

_NET_WM_WINDOW_TYPE_SPLASH indicates that the window is a splash screen displayed as an application is starting up.

_NET_WM_WINDOW_TYPE_DIALOG indicates that this is a dialog window. If _NET_WM_WINDOW_TYPE is not set, then managed windows with WM_TRANSIENT_FOR set MUST be taken as this type. Override-redirect windows with WM_TRANSIENT_FOR, but without _NET_WM_WINDOW_TYPE must be taken as _NET_WM_WINDOW_TYPE_NORMAL. 
```

**Comment by osleg** [comment]
all3 3 should be float by default

**Comment by esjeon** [comment]
The feature is implemented with commit 68b7df3.
